src/main/java/org/jnode/fs/xfs/directory/BlockDirectoryEntry.java

```diff
@@ -1,36 +1,74 @@
 package org.jnode.fs.xfs.directory;
 
+import lombok.Getter;
 import org.jnode.fs.util.FSUtils;
 import org.jnode.fs.xfs.XfsObject;
 
 /**
  * A XFS block directory entry inode.
+ * <p>
+ * Space inside the directory block can be used for directory entries or unused entries. This is signified via a union of
+ * the two types:
+ * <pre>
+ *     typedef union {
```

Comment:
I was thinking that structurally this could be broken up into a base class or interface `BlockDirectoryEntry` with children/implementors `BlockDirectoryDataEntry` and `BlockDirectoryUnusedEntry`. That could look cleaner but it could also be structural overkill.